Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +128,310 @@ public static class CustomEntitySearch
+                            prevWhiteSpaceIndex = currTextCharIndex;
+                        // Skip past extra delineating characters in the front of the word in the text
+                        if (currWordCharIndex == 0 && currMismatch >= 0 && textCharArray[currTextCharIndex].IsDelineating())
+                        {
+                            currTextCharIndex++;
+                            wordFound.GetStringBuilder().Clear();
+                            currMismatch = 0;
+                            continue;
+                        }
+
+                        if (currMismatch <= leniency)
+                        {
+                            if (wordCharArray[currWordCharIndex] == textCharArray[currTextCharIndex])
+                            {
+                                if (textCharArray[currTextCharIndex].IsDelineating() &&
+                                    (bestMismatchPreWhitespace == -1 || bestMismatchPreWhitespace > currMismatch + (wordCharArray.Length - currWordCharIndex)))
+                                {
+                                    whitespaceOffset = currTextCharIndex;
+                                    bestMismatchPreWhitespace = currMismatch + (wordCharArray.Length - currWordCharIndex);
+                                }
+                                wordFound.Write(text[currTextCharIndex]);
+                                currTextCharIndex++;
+                                currWordCharIndex++;
+                            }
+                            else
+                            {
+                                double potTextMismatch = (textCharArray[currTextCharIndex].IsAccent()) ? 0.5 : 0;
+                                double potWordMismatch = (wordCharArray[currWordCharIndex].IsAccent()) ? 0.5 : 0;
+                                // fuzzy situation?
+                                // accent case adds 0.5
+                                if (String.Compare(wordCharArray[currWordCharIndex].ToString(), textCharArray[currTextCharIndex].ToString(),
+                                    CultureInfo.CurrentCulture, CompareOptions.IgnoreNonSpace) == 0)
+                                {
+                                    currMismatch += 0.5;
+                                    wordFound.Write(text[currTextCharIndex]);
+                                    currTextCharIndex++;
+                                    currWordCharIndex++;
+                                }
+                                else if (potWordMismatch > 0 && potTextMismatch == 0)
+                                {
+                                    currMismatch += 0.5;
+                                    currWordCharIndex++;
+                                }
+                                else if (potWordMismatch == 0 && potTextMismatch > 0)
+                                {
+                                    currMismatch += 0.5;
```

Comment:
>0.5 [](start = 52, length = 3)

you're not using potWordMismatch as a double for these conditionals since you're just checking true/false and hardcoding currMismatch offset.

Can you change these conditionals to check textIsAccent and wordIsAccent booleans instead of this ternary double assignment?

Reply:
also, is it possible to consolidate this logic with the diffing logic below? The only difference is that accents are a different point value?

---
In reply to: [319582775](https://github.com/Azure-Samples/azure-search-power-skills/pull/5#discussion_r319582775) [](ancestors = 319582775)